.github/workflows/pr-gate.yml

```diff
@@ -0,0 +1,70 @@
+name: pr-gate
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  unit-test:
+    if: github.repository_owner == 'keikoproj'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
+      - name: Download Kubebuilder
+        run: |
+          curl -L -o kubebuilder_2.3.2_linux_amd64.tar.gz https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.3.2/kubebuilder_2.3.2_linux_amd64.tar.gz
+          tar -zxvf  kubebuilder_2.3.2_linux_amd64.tar.gz
+          sudo mv kubebuilder_2.3.2_linux_amd64 /usr/local/kubebuilder
+
+      - name: Test
+        run: |
+          make test
+
+      - name: Upload to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          file: ./coverage.txt
+
+  buildx:
+    if: github.repository_owner == 'keikoproj'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
+      - name: Download Kubebuilder
+        run: |
+          curl -L -o kubebuilder_2.3.2_linux_amd64.tar.gz https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.3.2/kubebuilder_2.3.2_linux_amd64.tar.gz
+          tar -zxvf  kubebuilder_2.3.2_linux_amd64.tar.gz
+          sudo mv kubebuilder_2.3.2_linux_amd64 /usr/local/kubebuilder
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: arm64,amd64
```

Comment:
Do we want to include arm, with arm64 here?

Reply:
We will only build for `arm64` and `amd64`

Reply:
Fyi. I created a new issue for pushing, here we're just runnig tests & builds. This issue is still TBD to address [Pushes and Multi-arch](https://github.com/keikoproj/addon-manager/issues/106).